src/app/loadout-builder/process-worker/process-utils.ts

```diff
@@ -207,7 +207,10 @@ export function canTakeSlotIndependantMods(
           item.energy.val + generalEnergy.val + otherEnergy.val <= item.energy.capacity &&
           (item.energy.type === generalEnergy.type ||
             generalEnergy.type === DestinyEnergyType.Any ||
-            item.energy.type === DestinyEnergyType.Any);
+            item.energy.type === DestinyEnergyType.Any) &&
+          (otherEnergy.type === generalEnergy.type ||
+            generalEnergy.type === DestinyEnergyType.Any ||
+            otherEnergy.type === DestinyEnergyType.Any);
```

Comment:
I would love to pull this out into a function but I am worried about performance with function call stacks.